content/blog/2020-05-05-updates-to-ragg-and-systemfonts/index.Rmarkdown

```diff
@@ -0,0 +1,169 @@
+# plot using jpeg
+jpegfile <- path(knitr::fig_path(),  "jpeg_image.jpeg")
+agg_jpeg(jpegfile, 2000, 2000)
+plot(noise, z)
+invisible(dev.off())
+
+knitr::include_graphics(jpegfile)
+```
+
+As you can hopefully agree on, there is no noticeable quality difference between the two. What about file size?
+
+```{r}
+file_size(pngfile)
+file_size(jpegfile)
+```
+
+We observe more than a 50% file size reduction. Again, bear in mind that this is only when saving image-type plots with high-frequency variations. `agg_png()` should always be your go-to device for any standard type of data visualization.
+
+## systemfonts 0.2.0
+systemfonts is a package that spun out of the development of ragg. Its main claim-to-fame is its ability to find the correct font file for a specific font and style request. With this release we have fixed up a lot of the underlying implementation as well as added a bunch of new features that may interest you.
+
+### Why are text so hard?
+Before we delve into the features of systemfonts it may be good to understand why fonts doesn't just work. R's font support has always been somewhat lacking and a range of packages has tried to solve this, e.g. extrafont and sysfonts. At its core text and fonts are just very messy, with differences between operating systems and font file formats to name some of the challenges. For a humorous overview consult the [*Text Rendering Hates You*](https://gankra.github.io/blah/text-hates-you/) article and have a chuckle. There are simply so many issues related to how finding the font files, reading the content, figuring out the correct glyphs for a character string (which may be encoded as god knows what) and determining the correct position of each glyph, that most applications that does something with text uses a high-level solutions such as those provided by the operating system or e.g. [Pango](https://www.pango.org). With all these complications it is easy to forgive the graphic devices in R for not being complete, since they have to take care of all of this from scratch.
+
+### New features
+The core functionality of systemfonts (finding font files) has been expanded with this release. The package makes heavy use of caching as font queries may happen many hundred times during rendering of a single plot. This means that if a font is installed while your R session is running it may not be picked up. systemfonts now provides a `reset_font_cache()` function to flush the cache so that newly installed fonts will be detected. It has also gained the ability to register fonts that are not installed on the system so that e.g. packages can provide a new set of fonts. This is done with the `register_font()` function. The font registration can also be
+used to get access to fonts that are not easily targeted with the limited information you can pass through the graphic engine in R (family, bold-or-not, italic-or-not), or if you want to overwrite the system-specific mappings of `'sans'`, `'serif'`, and `'mono'`.
```

Comment:
```suggestion
The core functionality of systemfonts (finding font files) has been expanded with this release. It now makes heavy use of caching to speed up the font queries that may happen many hundreds of times for a single plot. This gives a nice speed boost, but comes at a cost: if a font is installed while your R session is running, you'll need to explicitly run `systemfonts::reset_font_cache()` flush the cache. 

systemfonts has also gained the ability to register fonts that are not installed in the usual place so that e.g. packages can provide a new set of fonts. This is done with the `register_font()` function. The font registration can also be used to get access to fonts that are not easily targeted with the limited information you can pass through the graphic engine in R (family, bold-or-not, italic-or-not), or if you want to overwrite the system-specific mappings of `'sans'`, `'serif'`, and `'mono'`.
```